keyboards/gmmk/pro/rev1/ansi/keymaps/vnmm/keymap.c

```diff
@@ -0,0 +1,75 @@
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LALT, KC_LGUI,                            KC_SPC,                             KC_RGUI, WIN_F,   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [WIN_FN] = LAYOUT(
+        _______, DF_WIN,  DF_MAC,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          KC_MPLY,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        RGB_TOG, RGB_MOD, RGB_VAI, RGB_SPI, RGB_HUI, RGB_SAI, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, RGB_RMOD,RGB_VAD, RGB_SPD, RGB_HUD, RGB_SAD, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______,          _______, _______, _______,
+        EE_CLR,  _______, _______,                            QK_BOOT,                            _______, _______, _______, KC_MPRV, _______, KC_MNXT
+    ),
+
+    [MAC_FN] = LAYOUT(
+        _______, DF_WIN,  DF_MAC,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          KC_MPLY,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        RGB_TOG, RGB_MOD, RGB_VAI, RGB_SPI, RGB_HUI, RGB_SAI, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, RGB_RMOD,RGB_VAD, RGB_SPD, RGB_HUD, RGB_SAD, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______,          _______, _______, _______,
+        EE_CLR,  _______, _______,                            QK_BOOT,                            _______, _______, _______, KC_MPRV, _______, KC_MNXT
+    ),
+};
+
+void matrix_init_user(void) {
+#ifdef RGB_MATRIX_ENABLE
+    rgb_matrix_init_user();
+#endif
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    //return true; //set to return false to counteract enabled encoder in pro.c
+    return false;
+}
```

Comment:
Should replace this with the [encoder map](https://docs.qmk.fm/#/feature_encoders?id=encoder-map) feature, which is easier to write and supports VIA remapping too